shared/js/initApp.js

```diff
@@ -61,17 +61,22 @@ var baseOptions = {
     }
   },
   showInstructionsWrapper: function(showInstructions) {
-    var hasInstructions = appOptions.level.instructions || appOptions.level.aniGifURL;
-    if (!hasInstructions || this.share || appOptions.level.skipInstructionsPopup) {
+    // Always skip all pre-level popups on share levels or when configured thus
+    if (this.share || appOptions.level.skipInstructionsPopup) {
       return;
     }
 
-    if (appOptions.autoplayVideo) {
+    var hasVideo = !!appOptions.autoplayVideo;
+    var hasInstructions = appOptions.level.instructions || appOptions.level.aniGifURL;
```

Comment:
Any reasoning for using !! on the first but not the second? Seems like with our without should be valid in both cases, but might want to be consistent.